client/src/components/util/truncate.js

```diff
@@ -0,0 +1,78 @@
+import React, { cloneElement } from "react";
+import { Tooltip, Position } from "@blueprintjs/core";
+
+import { tooltipHoverOpenDelayQuick } from "../../globals";
+
+export default (props) => {
+  const { children } = props;
+  // Truncate only support a single child with a text child
+
+  if (
+    React.Children.count(children) === 1 &&
+    React.Children.count(children.props?.children) === 1
+  ) {
+    const originalString = children.props.children;
+
+    const firstString = originalString.substr(0, originalString.length / 2);
+    const secondString = originalString.substr(originalString.length / 2);
+
+    const splitStyle = {
+      ...children.props.style,
+      display: "flex",
+      overflow: "hidden",
+      justifyContent: "flex-start",
+    };
+
+    const firstStyle = {
```

Comment:
Since the style objects are constant, we can lift them out of the function declaration, so JS doesn't need to re-instantiate a new object every time this function is called.

E.g.,

```ts
const SPLIT_STYLE = {
  display: "flex",
  overflow: "hidden",
  justifyContent: "flex-start",
}

const FIRST_STYLE = {
  overflow: "hidden",
      textOverflow: "ellipsis",
      whiteSpace: "nowrap",
      flexShrink: 1,
      minWidth: '5px', 
}

const SECOND_STYLE_INNER = {
  position: "absolute",
  right: 0,
  color: "initial",
}

const SECOND_STYLE = {
  color: "transparent",
  position: "relative",
  overflow: "hidden",
  whiteSpace: "nowrap",
}

...

export default (props) => {
    ...

   const splitStyle = { ...children.props.style, ...SPLIT_STYLE }

    const truncatedJSX = (
      <span style={splitStyle}>
        <span style={FIRST_STYLE}>{firstString}</span>
        <span style={SECOND_STYLE}>
          {secondString}
          <span style={SECOND_STYLE_INNER}>{secondString}</span>
        </span>
      </span>
    );
})


```